src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/AddPackageCommandUtility.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+namespace NuGet.CommandLine.XPlat.Utility
+{
+    public static class AddPackageCommandUtility
+    {
+        /// <summary>
+        /// Return the latest version available in the sources
+        /// </summary>
+        /// <param name="sources">Sources to look at</param>
+        /// <param name="logger">Logger</param>
+        /// <param name="packageId">Package to look for</param>
+        /// <param name="prerelease">Whether to include prerelease versions</param>
+        /// <returns>Return the latest version available from multiple sources and if no version is found returns null.</returns>
+        public static async Task<NuGetVersion> GetLatestVersionFromSources(IList<PackageSource> sources, ILogger logger, string packageId, bool prerelease)
```

Comment:
It's super easy to forget to `await` a task in C#, so style guide is to suffix async methods with `Async`:

```suggestion
        public static async Task<NuGetVersion> GetLatestVersionFromSourcesAsync(IList<PackageSource> sources, ILogger logger, string packageId, bool prerelease)
```